### PR TITLE
State sync for physics

### PR DESCRIPTION
**Pull request by hamzzam**
## Summary

The base for state sync is complete. Linear/angular velocity is now synced across the network.

## Checklist
- [ ] Pre-push checks pass `npm run check`
  - [ ] Linter passing via `npm run lint`
  - [ ] Unit & Integration tests passing via `npm run test:packages`
  - [ ] Docker build process passing via `npm run build-client`
- [ ] If this PR is still a WIP, convert to a draft 
- [ ] When this PR is ready, mark it as "Ready for review"
- [ ] Changes have been manually QA'd
- [ ] Changes reviewed by at least 2 approved reviewers

## References

_References to pertaining issue(s)_

## QA Steps

1. `git checkout pr_branch_name`
2. `npm install`
3. `npm run dev-reinit`
4. `npm run dev`

List any additional steps required to QA the changes of this PR, as well as any supplemental images or videos.

## Reviewers

@HexaField @speigg

### CODE REVIEW

File: packages/client-core/src/components/Debug/index.tsx [hunk 1]
```diff
@@ -25,6 +27,16 @@ export const Debug = () => {
           toggleAvatarDebug()
         }
       }
+
+      if (ev.key === 'q') {
+        if (document.activeElement?.querySelector('canvas')) {
```
Comment (HexaField) [inline packages/client-core/src/components/Debug/index.tsx:32]: lets move this to a keybind in AvatarInputSchema and surround it with an `if (isDev)` check

File: packages/engine/src/physics/components/VelocityComponent.ts [hunk 2]
```diff
@@ -9,13 +9,15 @@ import { createMappedComponent } from '../../ecs/functions/ComponentFunctions'
  */
 
 export type VelocityComponentType = {
-  velocity: Vector3
+  linearVelocity: Vector3
```
Comment (HexaField) [inline packages/engine/src/physics/components/VelocityComponent.ts:12]: we should rename these to `linear` and `angular` as the velocity word is redundant, makes it easier to read

File: packages/projects/default-project/PhysicsSimulationTestSystem.ts [hunk 3]
```diff
@@ -93,14 +105,18 @@ export const generateSimulationData = (numOfObjectsToGenerate) => {
   // TODO: Generating the object as a network object here may result in an error sometimes,
   // the reason being the spawn object network action from server is received before the object has been added to the scene locally.
   // In order to test network functionality, call the generateSimulationData directly from scene loading logic in code for now.
-  generatePhysicsObject(config, new Vector3(0, 15, 0))
+  generatePhysicsObject(config, new Vector3(0, 15, 0), true)
 }
 
 const defaultSpawnPosition = new Vector3()
+const defaultScalePosition = new Vector3(1, 1, 1)
```
Comment (HexaField) [inline packages/projects/default-project/PhysicsSimulationTestSystem.ts:112]: should be named `defaultScale`

File: packages/projects/default-project/PhysicsSimulationTestSystem.ts [hunk 4]
```diff
@@ -133,15 +149,16 @@ export const generatePhysicsObject = (
 
   // Add empty model node
   const entity = createEntity()
-  let entityTreeNode = new EntityTreeNode(entity, getUUID())
+  const uuid = getUUID()
+  let entityTreeNode = new EntityTreeNode(entity, uuid)
```
Comment (HexaField) [inline packages/projects/default-project/PhysicsSimulationTestSystem.ts:153]: prefer const over let when possible

**Review by HexaField — CHANGES_REQUESTED** [review]

File: packages/projects/default-project/PhysicsSimulationTestSystem.ts (hunk 3 above)
Comment (hamzzam) [reply packages/projects/default-project/PhysicsSimulationTestSystem.ts:112]: lol, copy-paste really should be avoided 😄

**Review by HexaField — COMMENTED** [review]
LGTM!

**Pull request merged**